snuba/subscriptions/store.py

```diff
@@ -39,39 +39,38 @@ class RedisSubscriptionStore:
     defined by the `key` constructor param.
     """
 
-    KEY_TEMPLATE = "subscriptions:{}"
+    KEY_TEMPLATE = "subscriptions:{}:{}"
 
-    def __init__(self, client: RedisClientType, key: str):
+    def __init__(self, client: RedisClientType, dataset: Dataset, key: str):
         self.client = client
+        self.dataset = dataset
         self.key = key
         self.codec = SubscriptionCodec()
```

Comment:
It'd probably be good to loop back and make these private later, but better for them to be consistent right now.